### PR TITLE
Removed RecordStore#put(Map.Entry<Data, Object> entry) method.

### PR DESCRIPTION
**Pull request by ahmetmircik**
Instead of it RecordStore#put(key,value,ttl) should be used, they are doing same job.

### CODE REVIEW

**Comment by ahmetmircik** [comment]
verify-parallel

File: hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/RecordStore.java [hunk 1]
```diff
@@ -292,6 +292,8 @@
      **/
     void loadAll(boolean replaceExistingValues);
 
-    /** Performs initial loading from a MapLoader if it has not been done before  **/
+    /**
+     * Performs initial loading from a MapLoader if it has not been done before
+     **/
     void maybeDoInitialLoad();
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/RecordStore.java]: Although not part of this PR; perhaps rename this method to 'ensureLoaded' ?

File: hazelcast/src/main/java/com/hazelcast/map/impl/operation/EntryOperation.java [hunk 2]
```diff
@@ -211,8 +211,8 @@ private EntryEventType pickEventTypeOrNull(Map.Entry entry) {
         return null;
     }
 
-    private void put(Object value) {
-        recordStore.put(new AbstractMap.SimpleImmutableEntry<Data, Object>(dataKey, value));
+    private void put(Data dataKey, Object value) {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/map/impl/operation/EntryOperation.java]: Why not call it key? From the context we know it is Data.

**Comment by pveentjer** [comment]
Minor comments. So :+1:

File: hazelcast/src/main/java/com/hazelcast/map/impl/operation/EntryOperation.java (hunk 2 above)
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/map/impl/operation/EntryOperation.java]: done.
